src/instructlab/model/backends/llama_cpp.py

```diff
@@ -203,3 +186,70 @@ def read_root():
     if queue:
         queue.close()
         queue.join_thread()
+
+
+def augment_chat_template(
+    chat_template: str,
+    model_family: str,
+    model_path: pathlib.Path,
+    queue: Optional[multiprocessing.Queue],
+):
+    if chat_template == CHAT_TEMPLATE_TOKENIZER:
+        # llama_cpp_python calculates from the tokenizer config on load, so
+        # nothing to do
+        return
+
+    if chat_template == CHAT_TEMPLATE_AUTO:
+        # Currently "auto" maps to replacing with ilab stored templates
+        template, eos_token, bos_token = get_model_template(model_family, model_path)
+    else:
```

Comment:
Can you add a comment indicating that this third condition means `chat_template` is a template file?

Reply:
Also, this conditional block serves as a validation for `chat_template` should we extract it in a `validate_chat_template` and call it earlier (to fail early if needed)?

Reply:
Since both vllm and llama cpp log the chat template and error properly I preferred just delegating to them and avoid the double parsing.

Reply:
Souns good, just a small note about the third case 🙏🏻 so it's easier to understand for whoever is reading at the code and not familiar with it.